plugins/woocommerce/client/legacy/js/frontend/utils/custom-place-order-button.js

```diff
@@ -0,0 +1,241 @@
+
+	/**
+	 * Get the current form element based on page context.
+	 *
+	 * @return {jQuery} The form element.
+	 */
+	function getForm() {
+		if ( $( 'form.checkout' ).length ) {
+			return $( 'form.checkout' ).first();
+		}
+		if ( $( '#order_review' ).length ) {
+			return $( '#order_review' ).first();
+		}
+		if ( $( '#add_payment_method' ).length ) {
+			return $( '#add_payment_method' ).first();
+		}
+		return $( [] );
+	}
+
+	/**
+	 * Get a list of gateway IDs with custom place order buttons from server config.
+	 *
+	 * @return {Array} List of gateway IDs
+	 */
+	function getGatewaysWithCustomButton() {
+		// Try multiple param sources for compatibility across pages
+		if ( typeof wc_checkout_params !== 'undefined' && wc_checkout_params.gateways_with_custom_place_order_button ) {
+			return wc_checkout_params.gateways_with_custom_place_order_button;
+		}
+		if ( typeof wc_add_payment_method_params !== 'undefined' && wc_add_payment_method_params.gateways_with_custom_place_order_button ) {
+			return wc_add_payment_method_params.gateways_with_custom_place_order_button;
+		}
+		return [];
+	}
+
+	/**
+	 * Check if a gateway has a custom place order button registered via a server-side flag.
+	 *
+	 * @param {string} gatewayId - The payment gateway ID
+	 * @return {boolean} True if gateway has custom button
+	 */
+	function gatewayHasCustomPlaceOrderButton( gatewayId ) {
+		return getGatewaysWithCustomButton().indexOf( gatewayId ) !== -1;
+	}
+
+	/**
+	 * Create or get the container for custom place order buttons.
+	 * The container is scoped to the current form to handle multiple forms on a page.
+	 * We're not creating the container server-side to avoid introducing a new template
+	 * that might break compatibility with subscriptions or other extensions.
+	 *
+	 * @return {jQuery} The container element
+	 */
+	function getOrCreateCustomButtonContainer() {
+		if ( $customButtonContainer && $customButtonContainer.length && $.contains( document, $customButtonContainer[ 0 ] ) ) {
+			return $customButtonContainer;
+		}
+
+		$customButtonContainer = $( '<div class="wc-custom-place-order-button"></div>' );
+		getForm().find( '#place_order' ).after( $customButtonContainer );
+
+		return $customButtonContainer;
+	}
+
+	/**
+	 * Remove the custom button container.
+	 */
+	function removeCustomButtonContainer() {
+		if ( $customButtonContainer && $customButtonContainer.length ) {
+			$customButtonContainer.remove();
+			$customButtonContainer = null;
+		}
+	}
+
+	/**
+	 * Clean up the current custom button if any.
+	 */
+	function cleanupCurrentCustomButton() {
+		if ( activeCustomButtonGateway && customPlaceOrderButtons[ activeCustomButtonGateway ] ) {
+			try {
+				customPlaceOrderButtons[ activeCustomButtonGateway ].cleanup();
+			} catch ( e ) {
+				// Log errors to help gateway developers debug their cleanup implementation.
+				// eslint-disable-next-line no-console
+				console.error( 'Error in custom place order button cleanup:', e );
```

Comment:
I considered _not_ erroring, or using `wc_checkout_params.debug_mode`. But (for example) in files like `address-autocomplete.js` and `address-autocomplete-common.js` the errors are _also_ directly logged.